### PR TITLE
return complete response if asked

### PR DESCRIPTION
**Pull request by mkoura**
Currently it's not possible to get response data like HTTP status.
This PR proposes a way how to access such data and maintain backwards compatibility.

Example:
```
In [4]: rest_api.collections.requests.get(id=25)
Out[4]: <Entity u'https://10.8.199.218/api/automation_requests/25'>

In [5]: rest_api.response
Out[5]: <Response [200]>
```

### CODE REVIEW

**Comment by mkoura** [comment]
@simon3z @cben please review

**Comment by simon3z** [comment]
@dkorn @cben please review

**Comment by mkoura** [comment]
@dkorn @cben any comments?

**Comment by cben** [comment]
Sorry, saw the PR, wanted to think, but I should have said something.

1. This will still raise exceptions (and not let you access http response) on http errors (?), bad json, and "error" field in json.  Is that OK for your purposes?

2. Another approach would be defining existing methods in terms of new method(s) doing less.
    Imagine API like this:
    ```
    response = client.raw_post(url, foo='bar')
    # can now inspect `respose.status_code` etc.

    # So far no parsing/processing happened.  If you want it, call either of:
    response.json()
    response.process()
    ```
    then `client.post(...)` would be alias for `client.raw_post(...).process()`.

    - This way we could also drop the empty-response special casing in post() and delete(), just use `raw_post()` without parsing where you don't expect a body...  However this would be separate, backward-incompatible, change, and not sure it's worth the trouble.  @dkorn any opinion on this?

    Having 4 raw_get, raw_post etc methods might be overkill for rare lower-level usage.
    Could do with one `client.request(method, url, ...)` used by all higher-level `.get()`, `.post()` etc.

I don't feel strongly about this.  (2) is just stylistic.  I'm slightly concerned with (1) — could we later discover a need to get raw response before raising errors?  Growing yet another flag / api level would be awkward...
LGTM if you believe it's fine.

**Comment by mkoura** [comment]
@cben unfortunately num 2 would mean that we would have to do all additional processing separately, if we would like to do things like
```
result = api.collections.requests.action.edit(...)
check_response()
result[0].action.approve(...)
```

How about just storing the last response in instance variable, like this:
```
In [4]: rest_api.collections.requests.get(id=25)
Out[4]: <Entity u'https://10.8.199.218/api/automation_requests/25'>

In [5]: rest_api.response
Out[5]: <Response [200]>
```
That would be backwards compatible and sufficient for my purposes. See my last commit.

File: src/manageiq_client/api.py
Comment (cben) [inline src/manageiq_client/api.py]: Can you explain this one, and the one in `Entity.reload()`?
I see `reload()` are called implicitly from many places, I suppose you don't want `api.response` changed unpredictably?
OTOH there are things like `entity.exists()` that "obviously" make a request and not saving the response might be surprising.

What I mainly wonder, is this a "slightly nicer to use this way" thing, or an essential "would break otherwise" thing?
If nicer, LGTM.  If it's essential, that'd be a sign this approach is brittle and I'd prefer something more explicit— it seems very hard to me to follow all code paths and which response would be saved last in every case...

**Review by cben — APPROVED** [review]
Clever!
I see now, all 3 approaches have a price for code that wraps `.get()` and friends.  Your first approach requires repeating the `if full_response` dance at every level, my approach (2) may require duplicating methods at every level, and with this approach it's not always clear which response would be saved.
But overall pleasantly non-invasive.

You will likely be the only user of this functionality, so I trust your judgement.  LGTM.

File: src/manageiq_client/api.py
Comment (simon3z) [reply src/manageiq_client/api.py]: @mkoura @cben FYI I'll wait an answer / revision here before merging.

**Comment by cben** [comment]
LGTM++ love the new logic

File: src/manageiq_client/api.py
Comment (mkoura) [reply src/manageiq_client/api.py]: @cben You're right, it's hard to follow which response would be saved, I simplified it a bit.
My idea is that it's ok to save always the last response with the exception of "actions" (e.g. `api.collection.services.action.create()`). Here the user wants response from the action itself, not responses produced during result processing.

**Comment by mkoura** [comment]
@simon3z @cben I squashed the commits and updated description.

**Pull request merged**
